### ISSUE TITLE
Having hard time to understand how to tell Sparkle about the next builds

### ISSUE THREAD

**Issue by pradeepb28** [sparkle-project/Sparkle#2102]
## Summary

I'm new to Sparkle, this is my first project work on a mac app with Sparkle. I've so far successfully integrated Sparkle with test.xml from the Sparkle website. I'd like to go to the next step like how to tell Sparkle that there is a next build.
- What exactly to be done in xml file so I can replace it with the real one.( also I read in the docs to generate appcast xml file using a tool but I am not quite sure what tool it means.)
- In Xcode, it is quite straightforward to archive subsequent builds but what do I have to make sure Sparkle knows that I have a new build.

## Possible Fix

For time being, it'd great if there are detailed steps how to solve this. 

Thank you Sparkle team 🙏

**Comment by zorgiepoo** [comment]
Check

https://sparkle-project.org/documentation/#5-publish-your-appcast
https://sparkle-project.org/documentation/publishing/#publishing-an-update

The generate appcast tool is called `generate_appcast`. Try using it first, instead of opting into manual process. Updates need to be signed too.

A new update is done by adding a new `item` to the appcast whose `sparkle:version` matches your new `CFBundleVersion` and has an enclosure with your download link to the app.

Lastly this isn't well thought out request. You can at least paste what you have tried doing, what did or did not work, and which specific parts of the docs you did not understand. And be sure to critically read through the documentation rather than skim or skip parts. And then with all that, people may be able to assist better.

**Comment by pradeepb28** [comment]
Thank you. 
- I had trouble in understanding what Sparkle archive distribution folder means because I had integrated Sparkle using pods so it wasn't quite obvious where to find it. Now the problem is resolved.
- If I understand correctly appcast is generated based on the archive folder as well as the xml file path that is used in the SUFeedURL. But how to write the xml file path in the first place? (I did some tests with test.xml file from Sparkle but how do I create my own)
- For the future updates, do I have to upload the archive folder (zip & xml file) of that particulary update in a server?

![CleanShot 2022-03-13 at 11 18 41@2x](https://user-images.githubusercontent.com/6823101/158046929-8023d2a4-19af-40aa-8da6-4758fa021f13.png)

**Comment by zorgiepoo** [comment]
Can you actually try to create a folder with archives of your app and run generate_appcast on it like it says above and then see what it outputs in that folder?

Yes you’ll need to upload the files to your server. Generate_appcast should generate a appcast file and enclosures automatically so check what it does.

**Comment by pradeepb28** [comment]
>Can you actually try to create a folder with archives of your app and run generate_appcast on it like it says above and then see what it outputs in that folder?
Thank you. I did and it generated the appcast file with enclosure url. Interesting, I haven't uploaded the archive folder in to the server yet but it generated the enclosure url. (I suppose enclosure url is where the archive folder must be uploaded?)

**Comment by zorgiepoo** [comment]
Yes. Check `generate_appcast --help` for options if you want to customize stuff like the download or release notes URL prefix but otherwise there you go.
